backend/src/middleware/handleNotifications/handleNotificationsMiddleware.js

```diff
@@ -0,0 +1,156 @@
+import { UserInputError } from 'apollo-server'
+import extractMentionedUsers from './notifications/extractMentionedUsers'
+import extractHashtags from './hashtags/extractHashtags'
+
+const notifyUsers = async (label, id, idsOfUsers, reason, context) => {
+  if (!idsOfUsers.length) return
+
+  // Done here, because Neode validation is not working.
+  const reasonsAllowed = ['mentioned_in_post', 'mentioned_in_comment', 'comment_on_post']
+  if (!reasonsAllowed.includes(reason)) {
+    throw new UserInputError('Notification reason is not allowed!')
+  }
+  if (
+    (label === 'Post' && reason !== 'mentioned_in_post') ||
+    (label === 'Comment' && !['mentioned_in_comment', 'comment_on_post'].includes(reason))
+  ) {
+    throw new UserInputError('Notification fits not to reason!')
+  }
+
+  const session = context.driver.session()
+  const createdAt = new Date().toISOString()
+  let cypher
+  switch (reason) {
```

Comment:
Now, this cypher statement is getting messy. Shall I jump in an try to refactor?

Reply:
that would be great @roschaefer, but @Tirokk has changes to the branch to push up, could you wait some hours?

Reply:
No the code is pushed up. @roschaefer 
I and @mattwr18 would like to have pairing with you about this.
May @ogerly and @alina-beck like to join it as well?